daemon/k8s_watcher.go

```diff
@@ -835,9 +835,6 @@ func (d *Daemon) addCiliumNetworkPolicy(obj interface{}) {
 		return
 	}
 
-	// Delete an eventual existing rule with matching label
```

Comment:
I thought it was a leftover, do we need it? If yes, doesn't the `opts := AddOptions{Replace: true}` do the same?